shashi/feb11.js

```diff
@@ -0,0 +1,97 @@
+// Create a function that takes in a sentence and a character to find. Return an object of each word in the sentence, with the count of the specified character as the value.
+// Examples
+// findOccurrences("Hello World", "o") ➞ {
+//   "hello" : 1,
+//   "world" : 1
+// }
+// findOccurrences("Create a nice JUICY function", "c") ➞  {
+//   "create" : 1,
+//   "a" : 0,
+//   "nice" : 1,
+//   "juicy" : 1,
+//   "function" : 1
+// }
+// findOccurrences("An APPLE a day keeps an Archeologist AWAY...", "A") ➞ {
+//   "an" : 1,
+//   "apple" : 1,
+//   "a" : 1,
+//   "day" : 1,
+//   "keeps" : 0,
+//   "archeologist" : 1,
+//   "away..." : 2
+// }
+
+const findOccurrences = (str, letter) => {
+  //var words = splitWords(str);
+  return countLetter(splitWords(str), letter);
+}
+
+// function to split string words to an array
+const splitWords = (str) => {
+  var wordsArr = [];
+  var word = "";
+  for (var i = 0; i < str.length; i++) {
+    if (str[i] != " ") {
+      word += str[i];
+    } else {
+      wordsArr.push(word);
+      word = "";
+    }
+  }
+  wordsArr.push(word);
+  return wordsArr;
+}
+
+//function to count the given letter in each array word and add the word and count to an object.
+const countLetter = (words, letter) => {
+  var endObj = {};
+  var counter = 0;
+  words.forEach(word => {
+    for (let i = 0; i < word.length; i++) {
```

Comment:
hmm if you're going to use a forEach on line 49. you should use a forEach here too.

Reply:
forEach is array method right? Can we use for a string?